crates/ruff_linter/src/rules/ruff/rules/incorrectly_parenthesized_tuple_in_subscript.rs

```diff
@@ -0,0 +1,78 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::ExprSubscript;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for use or omission of parentheses around tuples in subscripts,
+/// depending on the setting [`lint.ruff.parenthesize-tuple-in-subscript`]. By default, the use of parentheses
+/// is considered a violation.
+///
+/// ## Why is this bad?
+/// It is good to be consistent and, depending on the codebase, one or the other
+/// convention may be preferred.
+///
+/// ## Example
+///
+/// ```python
+/// directions = {(0, 1): "North", (-1, 0): "East", (0, -1): "South", (1, 0): "West"}
+/// directions[(0, 1)]
+/// ```
+///
+/// Use instead (with default setting):
+///
+/// ```python
+/// directions = {(0, 1): "North", (-1, 0): "East", (0, -1): "South", (1, 0): "West"}
+/// directions[0, 1]
+/// ```
+
+#[violation]
+pub struct IncorrectlyParenthesizedTupleInSubscript {
+    prefer_parentheses: bool,
+}
+
+impl AlwaysFixableViolation for IncorrectlyParenthesizedTupleInSubscript {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        if self.prefer_parentheses {
+            format!("Use parentheses for tuples in subscripts.")
+        } else {
+            format!("Avoid parentheses for tuples in scubscripts.")
+        }
+    }
+
+    fn fix_title(&self) -> String {
+        if self.prefer_parentheses {
+            "Add parentheses around tuple in subscript.".to_string()
+        } else {
+            "Remove parentheses from tuple in subscript.".to_string()
+        }
```

Comment:
I think these could be slightly more concise, so that we're not just repeating the violation message:

```suggestion
        if self.prefer_parentheses {
            "Parenthesize the tuple.".to_string()
        } else {
            "Remove the parentheses.".to_string()
        }
```